src/app/rpmostree-clientlib.cxx

```diff
@@ -1244,6 +1244,85 @@ rpmostree_print_diff_advisories (GVariant *rpm_diff, GVariant *advisories, gbool
   return TRUE;
 }
 
+/* print "manifest-diff" */
+gboolean
+rpmostree_print_manifest_diff (GVariant *manifest_diff, guint maxkeylen, GError **error)
+{
+  if (!manifest_diff)
+    return TRUE; /* Nothing to 🖨️ */
+
+  int total_str = 0;
+  int total_size_str = 0;
+  int total_removed_str = 0;
+  int removed_size_str = 0;
+  int total_added_str = 0;
+  int added_size_str = 0;
+
+  g_auto (GVariantDict) manifest_diff_dict;
+  g_variant_dict_init (&manifest_diff_dict, manifest_diff);
+
+  g_autoptr (GVariant) total
+      = g_variant_dict_lookup_value (&manifest_diff_dict, "total", G_VARIANT_TYPE ("t"));
+  if (!total)
+    return glnx_throw (error, "Missing \"total\" key");
+
+  g_variant_get (total, "t", &total_str);
```

Comment:
OK I did some more digging and analysis, and I bet this is the problem combined with `int total_str = 0;`.

The `t` format expects to write to a 64 bit pointer, but `int` is [32 bits on most architectures](https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models).

Then what ends up happening here is we corrupt other variables on the stack, and because `manifest_diff_dict` is stack allocated, we overwrite what it points to and it becomes garbage.

Reply:
Unfortunately using GVariant from C is just full of wild type unsafety like this and it requires strict discipline.  In this particular case though, we can use `g_variant_get_uint64()` instead which gives type safety.

But yeah, Rust.